ckanext/xloader/plugin.py

```diff
@@ -251,18 +247,20 @@ def _remove_unsupported_resource_from_datastore(resource_id):
     Double check the resource format. Only supported Xloader formats should have datastore tables.
     If the resource format is not supported, we should delete the datastore tables.
     """
-    lc = ckanapi.LocalCKAN()
+    context = {"ignore_auth": True}
     try:
-        res = lc.action.resource_show(id=resource_id)
+        res = toolkit.get_action('resource_show')(context, {"id": resource_id})
     except toolkit.ObjectNotFound:
         log.error('Resource %s does not exist.' % res['id'])
         return
 
     if _should_remove_unsupported_resource_from_datastore(res):
-        log.info('Unsupported resource format "{}". Deleting datastore tables for resource {}'
-            .format(res.get(u'format', u'').lower(), res['id']))
+        log.info('Unsupported resource format "%s". Deleting datastore tables for resource %s'
+            % (res.get(u'format', u'').lower(), res['id']))
```

Comment:
Don't use % based interpolation, just pass the arguments to the logging function and let it interpolate them.

Reply:
https://docs.python.org/3/howto/logging.html#logging-variable-data

Reply:
Ohhh had no idea. Not exactly a python developer so still learning the pythonic stuff. Thanks! this is good to know. Logging calls have been updated to this (and did it in my ckanext-validation PR as well)